lib/multi.c

```diff
@@ -1385,10 +1352,8 @@ static CURLMcode Curl_multi_wait(struct Curl_multi *multi,
     }
   }
 
-#ifndef USE_WINSOCK
   if(ufds_malloc)
     free(ufds);
-#endif
   if(ret)
     *ret = retcode;
   if(!extrawait || nfds)
```

Comment:
This condition needs to be updated to also be true if just the WinSock event was triggered.